plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt

```diff
@@ -54,9 +56,15 @@ class ScaleBarImpl : ScaleBar, View {
     }
 
   override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+    val (viewWidth, viewHeight) = calculateWidthAndHeight()
+    setMeasuredDimension(viewWidth.toInt(), viewHeight.toInt())
+  }
+
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  internal fun calculateWidthAndHeight(): Pair<Float, Float> {
     val width = mapViewWidth * settings.ratio
     val height = settings.run { textBarMargin + textSize + height + (borderWidth * 2) }
-    setMeasuredDimension(width.toInt(), height.toInt())
+    return Pair(width, height)
```

Comment:
Historically spoken, custom drawing code should try avoid allocating new objects and reuse if possible. While this is less of an issue with hardware today. I do think it's an optimization that Google still preaches about today. Could we either quantify how often onMeasure is called when zooming? this could be checked with memory profiler to verify the allocations of Pair

Reply:
And in general, would love to see what happens when you zoom in and how much allocations/de-allocations happen. Maybe there are things we aren't thinking about today :)

Reply:
This will not be invoked during the custom drawing, only when updating settings.